ModPatches/Altered Carbon Ultratech Unleashed/Patches/Altered Carbon Ultratech Unleashed/AC_Hediff.xml

```diff
@@ -0,0 +1,227 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Patch>
+
+	<Operation Class="PatchOperationConditional">
+		<xpath>Defs/HediffDef[defName="AC_PainProtection"]/stages/li/statFactors</xpath>
+		<nomatch Class="PatchOperationAdd">
+			<xpath>Defs/HediffDef[defName="AC_PainProtection"]/stages/li</xpath>
+			<value>
+				<statFactors />
+			</value>
+		</nomatch>
+	</Operation>
+
+	<Operation Class="PatchOperationAdd">
+		<xpath>Defs/HediffDef[defName="AC_PainProtection"]/stages/li/statFactors</xpath>
+		<value>
+			<Suppressability>0.5</Suppressability>
+		</value>
+	</Operation>
+
+	<Operation Class="PatchOperationReplace">
+		<xpath>Defs/HediffDef[defName="AC_RogianArmBlade"]/comps/li[@Class="AlteredCarbon.HediffCompProperties_MeleeWeapon"]/tools</xpath>
+		<value>
+			<tools>
+				<li Class="CombatExtended.ToolCE">
+					<label>blade</label>
+					<capacities>
+						<li>Cut</li>
+					</capacities>
+					<power>25</power>
+					<armorPenetrationBlunt>2.16</armorPenetrationBlunt>
+					<armorPenetrationSharp>0.87</armorPenetrationSharp>
+					<cooldownTime>2</cooldownTime>
+					<alwaysTreatAsWeapon>true</alwaysTreatAsWeapon>
+					<linkedBodyPartsGroup>Blade</linkedBodyPartsGroup>
+					<soundMeleeHit>MeleeHit_BionicSlash</soundMeleeHit>
+					<soundMeleeMiss>MeleeMiss_BionicSlash</soundMeleeMiss>
+				</li>
+				<li Class="CombatExtended.ToolCE">
+					<label>blade</label>
+					<capacities>
+						<li>Stab</li>
+					</capacities>
+					<power>20</power>
+					<armorPenetrationBlunt>3.6</armorPenetrationBlunt>
+					<armorPenetrationSharp>3.6</armorPenetrationSharp>
+					<cooldownTime>2</cooldownTime>
+					<alwaysTreatAsWeapon>true</alwaysTreatAsWeapon>
+					<linkedBodyPartsGroup>Blade</linkedBodyPartsGroup>
+					<soundMeleeHit>MeleeHit_BionicSlash</soundMeleeHit>
+					<soundMeleeMiss>MeleeMiss_BionicSlash</soundMeleeMiss>
+				</li>
+			</tools>
+		</value>
+	</Operation>
+
+	<Operation Class="PatchOperationReplace" MayRequire="Ludeon.RimWorld.Royalty">
```

Comment:
While it functions correctly here, using a `MayRequire` on an Operation doesn't _really_ work properly. I can't tell you exactly how or why, but I poked around with it and found that it does weird things with the conditionality of the patch, despite not throwing any errors.

In short, we avoid doing so b/c it might break silently in some weird way. Correct way to do it is to either nest these operations in a PatchOpSequence so they're on an `li` and then use MayRequire, or just use PatchOpFindMod.

Reply:
Nice catch, I’ll change them to use patchop find mod in a bit.